eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/DisallowedWordsCheck.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DisallowedWordsCheck extends AbstractCheck {
+    private final String[] DEFAULT_VALUES = new String[] { "URL", "HTTP" };
+    private Set<String> disallowedWords = new HashSet<>(Arrays.asList(DEFAULT_VALUES));
+    private String errorMessage = "%s, All Public API Classes, Fields and Methods should follow " +
+        "Camelcase standards for the following words: %s.";
+    private boolean applyToPublic = true;
+
+    /**
+     * Adds words that Classes, Methods and Variables that should follow Camelcasing standards
+     * @param disallowedWords words that should follow normal Camelcasing standards
+     */
+    public final void setDisallowedWords(String... disallowedWords) {
+        if (this.disallowedWords != null) {
+            Collections.addAll(this.disallowedWords, disallowedWords);
+        }
```

Comment:
The `if` check here will never be null, as you're creating `disallowedWords` as a field, and that is what the `this.disallowedWords` refers to. I would suggest changing the code to something like this:

```java
    public final void setDisallowedWords(String... words) {
        if (words == null || words.length == 0) {
            this.disallowedWords.clear();
        }
        Collections.addAll(this.disallowedWords, words);
    }
```

Reply:
Yes I have this fixed locally I just have been waiting to fix anything that came from the review.